src/prpy/perception/chisel.py

```
@@ -0,0 +1,139 @@
+        @param robot Required by the PerceptionMethod interface
+        @param ignored_bodies A list of known kinbodies to be masked out
+        of the chisel mesh
+        """
+        env = robot.GetEnv()
+        
+        if ignored_bodies is None:
+            ignored_bodies = []
+
+        # Dictionary mapping ignored bodies to start transform
+        orig_bodies = {}
+
+        #Check if chisel kinbody in env
+        already_in_env = False
+        for body in env.GetBodies():
+            if body.GetName() == 'Chisel/full_mesh':
+                already_in_env = True
+                break
+
+        #Remove chisel kinbody if present
+        if already_in_env:
+            chisel_kb = env.GetKinBody('Chisel/full_mesh')
+            env.Remove(chisel_kb)
+
+        try:
+            from kinbody_helper import transform_from_or
+
+            # Add any ignored bodies to the camera
+            # We need to transform all these bodies from their pose
+            #   in OpenRAVE to the equivalent pose in TF so that
+            #   chisel can perform the appropriate masking
+            #   Then we will transform them all back after the server
+            #   performs the detection
+            for b in ignored_bodies:
+                if b not in self.camera_bodies:
+                    with env:
+                        orig_bodies[b] = b.GetTransform()
+                    transform_from_or(kinbody=b, 
+                                      detection_frame=self.detection_frame,
+                                      destination_frame=self.destination_frame,
+                                      reference_link=self.reference_link)
+                    self.camera.add_body(b)
+                    self.camera_bodies.append(b)
+
+            #Reset Chisel
+            if self.reset_detection_srv is None:
+                import rospy, chisel_msgs.srv
+                srv_nm = self.serv_ns+'/Reset'
+                rospy.wait_for_service(srv_nm)
```

I suggest passing a `timeout` here. That way we can raise an exception with an informative error message if the Chisel service is not running.